zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,110 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+My Approch
+As i don't know how zulip store data in which database or file i simply create a
+database for me so that i can store the length of json response in database ,when
+the new page or space  is created in confluence i get the the update using zulip
+rest api due to which there was a change in length in json data so i compare the
+length with database length if length is less  than update length then i update the
+database length and fetch the last update data from json response and send it to zulip
+stream.
+"""
+import os
+import sys
+import argparse
+import zulip
+import requests
+import json
+INSTRUCTIONS = r"""
+confluence-bot --config-file=~/.zuliprc --search="@nprnews,quantum physics"
+Send Confluence space and page  to a Zulip stream.
+To use this script:
+0. Use `Confluence rest api ` to  `Confluence `
+1. Set up Confluence authentication, as described below
+2. Set up a Zulip bot user and download its `.zuliprc` config file to e.g. `~/.zuliprc`
+3. Subscribe the bot to the stream that will receive Confluence updates (default stream: confluence)
+4. Test the script by running it manually, like this:
+5. Configure a crontab entry for this script. A sample crontab entry
+that will process spaces and pages every 5 minutes is:
+*/5 * * * * /usr/local/share/zulip/integrations/confluence/confluence [options]
+== Setting up Confluence authentications ==
+Run this on a personal or trusted machine, because your API key is
+visible to local users through the command line or config file.
+This bot uses OAuth to authenticate with Confluence. Please create a
+~/.zulip_confluencerc with the following contents:
+[confluence]
+username = example@.com
+api_token = xxxxxxxxxxxx
+In order to obtain a consumer key & secret, you must register a
+new application under your Confluence account:
+1 Log in to https://id.atlassian.com.
+2 Click API tokens, then Create API token.
+3 Use Copy to clipboard, and paste the token to your script, or elsewhere:
+"""
+f = open("paras.txt", "w+")
+f.write('0')
+f.close()
+def confluence_api() -> None:
+    # Use rest api to fetch data from confluence here
+    # In authentication we provide confluence api token and then username
+    params = ('type', 'page')
+    response = requests.get('https://paras112.atlassian.net/wiki/rest/api/content',
```

Comment:
Again, you are using hardcoded values. Not everyone has their Confluence hosted at that URL.

Reply:
Maybe the URL can also be added to .zulip_confluencerc so that it can be grabbed from there? This way the user won't have to modify the script and can just add the required information to .zulip_confluencerc

Reply:
@mayankmadan: I agree yes, let's do that. :)